commit 814e44507f9656babc7d499d5f25997e3fe9e817
Author: Yifeng Li <tomli@tomli.me>
Date:   Wed Feb 6 15:07:21 2019 +0800

    mips: loongson64: remove unreachable(), fix loongson_poweroff().
    
    commit 8a96669d77897ff3613157bf43f875739205d66d upstream.
    
    On my Yeeloong 8089, I noticed the machine fails to shutdown
    properly, and often, the function mach_prepare_reboot() is
    unexpectedly executed, thus the machine reboots instead. A
    wait loop is needed to ensure the system is in a well-defined
    state before going down.
    
    In commit 997e93d4df16 ("MIPS: Hang more efficiently on
    halt/powerdown/restart"), a general superset of the wait loop for all
    platforms is already provided, so we don't need to implement our own.
    
    This commit simply removes the unreachable() compiler marco after
    mach_prepare_reboot(), thus allowing the execution of machine_hang().
    My test shows that the machine is now able to shutdown successfully.
    
    Please note that there are two different bugs preventing the machine
    from shutting down, another work-in-progress commit is needed to
    fix a lockup in cpufreq / i8259 driver, please read Reference, this
    commit does not fix that bug.
    
    Reference: https://lkml.org/lkml/2019/2/5/908
    Signed-off-by: Yifeng Li <tomli@tomli.me>
    Signed-off-by: Paul Burton <paul.burton@mips.com>
    Cc: linux-mips@vger.kernel.org
    Cc: Huacai Chen <chenhc@lemote.com>
    Cc: Ralf Baechle <ralf@linux-mips.org>
    Cc: James Hogan <jhogan@kernel.org>
    Cc: linux-kernel@vger.kernel.org
    Cc: Aaro Koskinen <aaro.koskinen@iki.fi>
    Cc: stable@vger.kernel.org # v4.17+
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/arch/mips/loongson64/common/reset.c b/arch/mips/loongson64/common/reset.c
index a60715e11306..b26892ce871c 100644
--- a/arch/mips/loongson64/common/reset.c
+++ b/arch/mips/loongson64/common/reset.c
@@ -59,7 +59,12 @@ static void loongson_poweroff(void)
 {
 #ifndef CONFIG_LEFI_FIRMWARE_INTERFACE
 	mach_prepare_shutdown();
-	unreachable();
+
+	/*
+	 * It needs a wait loop here, but mips/kernel/reset.c already calls
+	 * a generic delay loop, machine_hang(), so simply return.
+	 */
+	return;
 #else
 	void (*fw_poweroff)(void) = (void *)loongson_sysconf.poweroff_addr;
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1329_814e44507f9656babc7d499d5f25997e3fe9e817.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1329_814e44507f9656babc7d499d5f25997e3fe9e817.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
